source/extensions/filters/http/jwt_authn/filter_factory.cc

```diff
@@ -46,8 +46,9 @@ FilterFactory::createFilterFactoryFromProtoTyped(const JwtAuthentication& proto_
   validateJwtConfig(proto_config);
   auto filter_config = std::make_shared<FilterConfig>(proto_config, prefix, context);
   return [filter_config](Http::FilterChainFactoryCallbacks& callbacks) -> void {
-    callbacks.addStreamDecoderFilter(
-        std::make_shared<Filter>(filter_config->stats(), Authenticator::create(filter_config)));
+    auto jwks_fetcher = Common::JwksFetcher::create(filter_config->cm());
```

Comment:
can we pass a lambda function as fetcher_factory so fetcher is only created when it is needed,  not always for each request?